### PR TITLE
Add Clock

### PR DESCRIPTION
**Pull request by Bolodya1997**
Adds `clock.Clock` interface for accessing time and `clock.Mock` for mocking time - #764.

### CODE REVIEW

**Review by denis-tingaikin — COMMENTED** [review]
I very like this direction :) 

@Bolodya1997 Should we put mocks into separate pkg to exclude them from cmd-*?

File: pkg/tools/clock/context.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+// WithClock wraps parent in a new context with Clock
+func WithClock(parent context.Context, clock Clock) context.Context {
+	if parent == nil {
+		panic("cannot create context from nil parent")
+	}
+	return context.WithValue(parent, clockKey, clock)
+}
+
+// GetClock returns Clock from context
+func GetClock(ctx context.Context) Clock {
```
[the remark this answers is not in the document]
Comment (Bolodya1997) [reply pkg/tools/clock/context.go]: Yep, looks better.

File: pkg/tools/clock/context.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	if rv, ok := ctx.Value(clockKey).(Clock); ok {
+		return rv
+	}
+	return new(clockImpl)
```
[the remark this answers is not in the document]
Comment (Bolodya1997) [reply pkg/tools/clock/context.go]: Fixed.

**Comment by Bolodya1997** [comment]
> @Bolodya1997 Should we put mocks into separate pkg to exclude them from cmd-*?

Yes, it is a good idea. Moved to `clockmock`.

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**